sw/device/tests/penetrationtests/firmware/sca/aes_sca.c

```diff
@@ -630,6 +630,129 @@ status_t handle_aes_sca_fvsr_key_start_batch_generate(ujson_t *uj) {
   return OK_STATUS();
 }
 
+static status_t trigger_aes_gcm(dif_aes_key_share_t key, dif_aes_iv_t iv,
+                                dif_aes_aad_t aad, dif_aes_data_t ptx,
+                                size_t ptx_num_valid_bytes, dif_aes_data_t *ctx,
+                                dif_aes_data_t *tag) {
+  // AES GCM configuration used for this test.
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeGcm,
+      .key_len = kDifAesKey128,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .manual_operation = kDifAesManualOperationManual,
+      .mask_reseeding = kDifAesReseedPer8kBlock,
+      .reseed_on_key_change = false,
+      .force_masks = false,
+      .ctrl_aux_lock = false,
+  };
+
+  // Write the initial key share, IV and data in CSRs.
+  TRY(dif_aes_start(&aes, &transaction, &key, &iv));
+  aes_manual_trigger();
+
+  // Load the AAD block into the AES.
+  // AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusIdle, true, 1000 * 1000);
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true,
+                                1000 * 1000);
+  TRY(dif_aes_load_aad(&aes, aad, true));
+  aes_manual_trigger();
+
+  // Load the PTX block into the AES.
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusIdle, true, 1000 * 1000);
+  TRY(dif_aes_set_gcm_text_mode(&aes, 16));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true,
+                                1000 * 1000);
+  TRY(dif_aes_load_data(&aes, ptx));
+  aes_manual_trigger();
+
+  // Read the CTX block from the AES.
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true,
+                                1000 * 1000);
+  TRY(dif_aes_read_output(&aes, ctx));
```

Comment:
Nit: you don't actually need to read the ciphertext. The tag is sufficient to verify correct operation :-)